v03_pipeline/lib/model/cached_reference_dataset_query.py

```diff
@@ -146,18 +147,30 @@ def query(self) -> Callable[[hl.Table, ReferenceGenome], hl.Table]:
         }[self]
 
     @classmethod
-    def for_dataset_type(
+    def for_reference_genome_dataset_type(
         cls,
+        reference_genome: ReferenceGenome,
         dataset_type: DatasetType,
     ) -> list['CachedReferenceDatasetQuery']:
-        return {
-            DatasetType.SNV_INDEL: [
+        crdqs = {
+            (ReferenceGenome.GRCh38, DatasetType.SNV_INDEL): [
                 CachedReferenceDatasetQuery.CLINVAR_PATH_VARIANTS,
                 CachedReferenceDatasetQuery.HIGH_AF_VARIANTS,
                 CachedReferenceDatasetQuery.GNOMAD_CODING_AND_NONCODING_VARIANTS,
                 CachedReferenceDatasetQuery.GNOMAD_QC,
             ],
-            DatasetType.MITO: [
+            (ReferenceGenome.GRCh38, DatasetType.MITO): [
+                CachedReferenceDatasetQuery.CLINVAR_PATH_VARIANTS,
+            ],
+            (ReferenceGenome.GRCh37, DatasetType.SNV_INDEL): [
```

Comment:
We actually have the same "cached_reference_dataset_queries" for 37 as we do for 38, so this change is purely for cosmetic consistency reasons.  For more context, the four queries we have are:
- clinvar_path_variants (used by hail search)
- high_af_variants (used by hail search)
- common coding and non-coding variants (used internvally by the sample type validation logic)
- gnomad qc (used by sex/relatedness for... something...)

Reply:
I see the advantage of the cosmetic change, but I think we would want the default behavior for new reference data sources that they be made available for both builds, so I feel like we would want some sort of shared list as the default place things go, and then that way you would need to make an explicit choice to exclude something for a given build. This could be as simple as defining a top-level constant for `SNV_INDEL_CRDQS` and then here having
```
(ReferenceGenome.GRCh38, DatasetType.SNV_INDEL): SNV_INDEL_CRDQS,
(ReferenceGenome.GRCh37, DatasetType.SNV_INDEL): SNV_INDEL_CRDQS,
```